### PR TITLE
"friend request" feature 

### PR DESCRIPTION
**Pull request by nadavshoshan**
1. DB- Django default DB is Sqlite, we saw online that in order to handle backend processing it is very recommended to change it to Postgresql. all the changes have been made on the givitsite/setting. solves #11 

2. first draft for getting item request from a friend:

I created a new app on our Django project called friendreq.

- model section: I added a new model called ItemRequest which defines the primary fields for our friend request (item, city, special requests).

- views: define a new function requestItem: when called using "GET" method show the user the request form. when called using "POST" method it will construct a new instance of ItemRequest object, update the DB and render a new HTML page which indicates that the request sent to the friend.

 solves #12

### CODE REVIEW

File: givitsite/friendreq/admin.py [hunk 1]
```diff
@@ -1,3 +1,5 @@
 from django.contrib import admin
-
+from .models import ItemRequest
 # Register your models here.
+
+admin.site.register(ItemRequest)
```
Comment (ifireball) [inline givitsite/friendreq/admin.py:5]: please fix the newline

**Review by ifireball — CHANGES_REQUESTED** [review]
The commit message is misleading - you're not adding a database, you're adding a model class, a route and a view. You're taking a vertical development approach which is good, but at this point I'd expect some separation between changes to different parts of the application, perhaps via different commits.

You seem to have made a decision to go from sqlite to Postgres - I expect to see details about this decision in the commit message and perhaps in the README as well.

I'm not seeing code anywhere to actually install and bring up postres itself - how did you test this code?

File: givitsite/friendreq/admin.py (hunk 1 above)
Comment (rotem-benzvi) [reply givitsite/friendreq/admin.py:5]: i will answer the 2 last issues- 
1. about the install postgres we testing everthing and update the bootstrap.sh with all change that needed you can see it in my pull request - https://github.com/beyond-io/givit/pull/9 we also upadate some guide needed in the readme.md. 2. about the decision to move to postgres i will add it to the readme.md..

last question- the seperation you exepect that we change this current PR or for the future? 

thanks 👍

Comment (ifireball) [reply givitsite/friendreq/admin.py:5]: Please make sure you explicitly state the dependency on PR #9 in the PR description so the PRs will be merged in the right order, unfortunately GitHub cannot enforce it (There are systems that can).

I expect the separation to happen for this PR. 
Here is how to do it with Git: https://embeddedartistry.com/blog/2018/02/19/code-cleanup-splitting-up-git-commits-in-the-middle-of-a-branch/

Comment (nadavshoshan) [reply givitsite/friendreq/admin.py:5]: Thanks for all the notes.
We will do it an re - pull

**Comment by nadavshoshan** [comment]
Hey, @ifireball I updated the pull request.
waiting for you review

thanks!

**Review by ifireball — CHANGES_REQUESTED** [review]
No seeing an improved PR description and title. The commit messages also should be improved - they include the *what* part but not the *why*. Please see commit and PR message guidelines in the SOW.

**Comment by nadavshoshan** [comment]
Hey, this is the description for the PR:

1. DB- Django default DB is Sqlite, we saw online that in order to handle backend processing it is very recommended to change it to Postgresql. all the changes have been made on the givitsite/setting. solves #11

2. first draft for getting item request from a friend:

> - I created a new app on our Django project called **friendreq**.

> - model section: I added a new model called **ItemRequest** which defines the primary fields for our friend request (item, city, special requests).

> - views: define a new function **requestItem**: when called using "GET" method show the user the request form. when called using "POST" method it will construct a new instance of **ItemRequest** object, update the DB and render a new HTML page which indicates that the request sent to the friend.

**Comment by ifireball** [comment]
You aught to be able to adit the PR description with the `edit` option in the `...` menu at the top. 

About the PR title - if you can't write it in a single short 3-7 word sentence it usually means you're doing too much in the same PR - In this case though, I think you can narrow it down to talking about the "friend request" feature - everything else can go in the bigger description text area.

Please keep greetings like "hey" or "good morning" out of PR descriptions and commit messages - use the chat for those.

**Pull request merged**